spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/Flow.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.core.dsl;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The AST node representing a flow. A flow is a series of jobs to execute sequentially. Those jobs
+ * can themselves be individual jobs or splits. In DSL form a flow is expressed like this:
+ * <pre><tt>aa || bb</tt></pre>.
+ *
+ * @author Andy Clement
+ */
+public class Flow extends LabelledComposedTaskNode {
+
+	private List<LabelledComposedTaskNode> series;
+
+	public Flow(List<LabelledComposedTaskNode> jobDefOrRefsWithConditions) {
+		super(jobDefOrRefsWithConditions.get(0).getStartPos(),
```

Comment:
I renamed it as the job name was wrong. I'm probably not going to assert though because it can't be null in the way it is called.